src/pages/ThorChainLP/Pool/components/PoolChart.tsx

```diff
@@ -1,34 +1,133 @@
 import { Button, ButtonGroup, Center, Flex, Stack } from '@chakra-ui/react'
-import { FaChartSimple } from 'react-icons/fa6'
-import { useTranslate } from 'react-polyglot'
-import { RawText } from 'components/Text'
+import { useQuery } from '@tanstack/react-query'
+import type { SingleValueData, UTCTimestamp } from 'lightweight-charts'
+import { useCallback, useMemo, useState } from 'react'
+import { reactQueries } from 'react-queries'
+import type { Interval } from 'react-queries/queries/midgard'
+import { SimpleChart } from 'components/SimpleChart'
+import { fromThorBaseUnit } from 'lib/utils/thorchain'
+import type {
+  MidgardSwapHistoryResponse,
+  MidgardTvlHistoryResponse,
+} from 'lib/utils/thorchain/lp/types'
+import { selectUserCurrencyToUsdRate } from 'state/slices/selectors'
+import { store } from 'state/store'
+
+const swapHistoryToChartData = (
+  swapHistory: MidgardSwapHistoryResponse | undefined,
```

Comment:
react-query return type implies no `undefined`, we should either update the query return value to reflect `undefined` being valid response or omit this imo

Reply:
https://github.com/shapeshift/web/pull/6624#discussion_r1556490870